### PR TITLE
Fix some minor issues in the Customizable infrastructure

### PR DESCRIPTION
**Pull request by mrambacher**
- Fix issue with OptionType::Vector when the nested item is a Customizable with no names
- Fix issue with OptionType::Vector to appropriately wrap the elements in a Vector;
- Fix an issue with nested Customizable object with a null immutable object still appearing in the mutable options;
- Fix/Add tests for null/empty customizable objects
- Move the RegisterTestObjects from customizable_test into testutil.

### CODE REVIEW

File: options/options_helper.cc [hunk 1]
```diff
@@ -1075,7 +1075,11 @@ Status OptionTypeInfo::Serialize(const ConfigOptions& config_options,
   } else if (IsCustomizable()) {
     const Customizable* custom = AsRawPointer<Customizable>(opt_ptr);
     if (custom == nullptr) {
-      *opt_value = kNullptrString;
```
Comment (zhichao-cao) [inline options/options_helper.cc:1078]: Add comment here.

**Review by zhichao-cao — APPROVED** [review]
LGMT, may need to mention you make GetOptionsMap public and slight behavior difference in NewSharedObject.

File: include/rocksdb/utilities/customizable_util.h [hunk 2]
```diff
@@ -223,23 +215,25 @@ template <typename T>
...
-      return Status::OK();
+      status = Status::OK();
+    } else if (status.ok()) {
+      status =
+          Customizable::ConfigureNewObject(config_options, *result, opt_map);
     }
-  }
-  if (!status.ok()) {
     return status;
+  } else if (opt_map.empty()) {
```
[the remark this answers is not in the document]
Comment (mrambacher) [reply include/rocksdb/utilities/customizable_util.h:232]: There is no ID and no map, meaning the user passed in an empty string to CreateFromString.  This scenario has been treated as a "clear the object".  If that is not the behavior we want/expect, I can see what happens if we change it
